controllers/csiaddonsnode_controller.go

```diff
@@ -39,17 +50,98 @@ type CSIAddonsNodeReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CSIAddonsNode object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
```

Comment:
this function now takes ~100 lines, consider making it simpler by adding helper functions

Reply:
> this function now takes ~100 lines, consider making it simpler by adding helper functions

its about 80 lines now, 
creating helper functions will need a lot of parameters(logger,instance,client...) to be passed to it, which makes it more ugly actually. 
(had tried it :sweat: )

On another hand, single flow will make it more readable

Reply:
lines 51 to 147... and that is without a lot of comments. I expect this function to grow over time, so some cleanup would be good.

It should be possible to handle different events in separate functions. If there are many parameters that need passing, consider adding a type like `CSIAddonsNodeOperation` containing the logger, client and other members. It can then have methods like `removeFinalizer()`,  `connect()` and so on.